Mainnet/DividendToken/DividendToken/DividendToken.cs

```diff
@@ -0,0 +1,288 @@
+﻿using Stratis.SmartContracts;
+using Stratis.SmartContracts.Standards;
+
+[Deploy]
+public class DividendToken : SmartContract, IStandardToken
+{
+
+    public ulong Dividends
+    {
+        get => PersistentState.GetUInt64(nameof(this.Dividends));
+        private set => PersistentState.SetUInt64(nameof(this.Dividends), value);
+    }
+
+    private Account GetAccount(Address address) => PersistentState.GetStruct<Account>($"Account:{address}");
+
+    private void SetAccount(Address address, Account account) => PersistentState.SetStruct($"Account:{address}", account);
+
+
+    public DividendToken(ISmartContractState state, ulong totalSupply, string name, string symbol)
+        : base(state)
+    {
+        this.TotalSupply = totalSupply;
+        this.Name = name;
+        this.Symbol = symbol;
+        this.SetBalance(Message.Sender, totalSupply);
+    }
+
+    /// <summary>
+    /// It is advised that deposit amount should to be evenly divided by total supply, 
+    /// otherwise small amount of satoshi may lost(burn)
+    /// </summary>
+    public override void Receive()
+    {
+        Dividends += Message.Value;
+    }
+
+    public bool TransferTo(Address to, ulong amount)
+    {
+        UpdateAccount(Message.Sender);
+        UpdateAccount(to);
+
+        return TransferTokensTo(to, amount);
+    }
+
+    public bool TransferFrom(Address from, Address to, ulong amount)
+    {
+        UpdateAccount(from);
+        UpdateAccount(to);
+
+        return TransferTokensFrom(from, to, amount);
+    }
+
+    Account UpdateAccount(Address address)
+    {
+        var account = GetAccount(address);
```

Comment:
Check that the account actually exists and is valid.

Reply:
It will return default value for Account struct which is expected behavior. Account value can't be invalid.